posix/get-buildtool-lang

```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+
+set -e
+
+# Define a list of language extensions
+detect_language() {
+    found_lang=""
+
+    # Define extensions for languages
+    case "$1" in
+        *.java) found_lang="Java" ;;
+        *.py) found_lang="Python" ;;
+        *.rb) found_lang="Ruby" ;;
+        *.go) found_lang="Golang" ;;
+        *.scala) found_lang="Scala" ;;
+        *.kt|*.kts) found_lang="Kotlin" ;;
+        *) found_lang="" ;;
+    esac
+    echo "$found_lang"
+}
+
+# Traverse the directory and detect languages
+found=""
+repo_path="./"
+for file in $(find "$repo_path" -type f 2>/dev/null | sort); do
```

Comment:
does this traverse each file in the repo? or the once in root directory?

Reply:
it will traverse each dir from root dir, as soon as a file is found it will return on best effort basis. It takes only few ms tested on a big repo